src/classes/RD2_ScheduleService_TEST.cls

```diff
@@ -949,9 +1016,281 @@ private with sharing class RD2_ScheduleService_TEST {
 
     }
 
+    /***
+    * @description Verifies visualization works with a large number of installments requested
+    */
+    @isTest
+    private static void shouldReturn480InstallmentsWhen480Requested() {
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact contact = getContact();
 
-    //// Helpers
+        npe03__Recurring_Donation__c rd =
+                TEST_RecurringDonationBuilder.constructEnhancedBuilder()
+                        .withContact(contact.Id)
+                        .withAmount(50)
+                        .withPaymentMethod('Credit Card')
+                        .withInstallmentPeriodMonthly()
+                        .withDayOfMonth('1')
+                        .withInstallmentFrequency(1)
+                        .withDateEstablished(Date.newInstance(2019, 11, 1))
+                        .withStartDate(Date.newInstance(2019, 11, 1))
+                        .build();
+
+        RD2_ScheduleService service = getScheduleServiceWithCurrentDateOverride(Date.newInstance(2019, 11, 1));
+
+        Test.startTest();
+        insert rd;
+        List<RD2_ScheduleService.Installment> installments = service.getNextDonationsForVisualization(rd, Date.newInstance(2019, 11, 1), 480);
+        Test.stopTest();
+
+        System.assertEquals(480, installments.size(), 'There should be 480 installments.');
+        System.assertEquals(Date.newInstance(2059, 10, 1), installments[479].nextDonationDate, 'Final installment date should be 10/1/2059.');
+
+    }
+
```

Comment:
What would you think about a test that returns a limited number of installments, based on an EndDate in the schedule? For example, if the Schedule record has a StartDate of 1/1/2020, and EndDate to 10/30/2020 and the method is called to retrieve 12 schedules, the unit test would verify that only 10 are returned.

Reply:
Sounds good.  If we do that, can't we say we already support fixed length recurring donations?

Reply:
Answering my own question...  Nope.  We'd need End Date on RD in order to make that claim.